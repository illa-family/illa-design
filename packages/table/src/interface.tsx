import {
  HTMLAttributes,
  ReactNode,
  TdHTMLAttributes,
  ThHTMLAttributes,
} from "react"
import { Column, Row, UseFiltersInstanceProps } from "react-table"
import { SerializedStyles } from "@emotion/react"
import { TableData } from "./table-data"

export type TableSize = "small" | "medium" | "large"

export type TableLayout = "auto" | "fixed"

export type TableAlign =
  | "left"
  | "center"
  | "right"
  | "start"
  | "end"
  | "flex-start"
  | "flex-end"

export interface TableProps<D extends TableData>
  extends HTMLAttributes<HTMLDivElement>,
    TableContextProps {
  columns?: ReadonlyArray<Column<D>>
  data?: readonly D[]
  tableLayout?: TableLayout
  bordered?: boolean
  disableSortBy?: boolean
  disableFilters?: boolean
  disableRowSelect?: boolean
  onRowSelectChange?: (rows: Array<Row<D>>) => void
  _css?: SerializedStyles
}

export interface TableFilterProps<D extends TableData> {
  renderFilterContent?: (columnProps: UseFiltersInstanceProps<D>) => ReactNode
  columnProps: UseFiltersInstanceProps<D>
  _css?: SerializedStyles
}

export interface TableContextProps {
  borderedCell?: boolean
  striped?: boolean
  showHeader?: boolean
  showFooter?: boolean
  size?: TableSize
  align?: TableAlign
}

export interface TBodyProps extends HTMLAttributes<HTMLTableSectionElement> {
  _css?: SerializedStyles
}

export interface TrProps extends HTMLAttributes<HTMLTableRowElement> {
  _css?: SerializedStyles
}

export interface TdProps
  extends Omit<TdHTMLAttributes<HTMLTableDataCellElement>, "align">,
    TableContextProps {
  _css?: SerializedStyles
}

export interface TFootProps extends HTMLAttributes<HTMLTableSectionElement> {
  _css?: SerializedStyles
}

export interface THeadProps extends HTMLAttributes<HTMLTableSectionElement> {
  _css?: SerializedStyles
}

export interface ThProps
  extends Omit<ThHTMLAttributes<HTMLTableHeaderCellElement>, "align">,
    TableContextProps {
  _css?: SerializedStyles
}
