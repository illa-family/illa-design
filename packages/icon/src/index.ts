export * from "./icon"
export * from "./interface"
export * from "./create-icon"
// icons
export * from "./icons/close"
export * from "./icons/person"
export * from "./icons/image-default"
export * from "./icons/loading"
export * from "./icons/copy"
export * from "./icons/checkmark"
export * from "./icons/eye-on"
export * from "./icons/eye-off"
export * from "./icons/error"
export * from "./icons/search"
export * from "./icons/link"
export * from "./icons/right"
export * from "./icons/error"
export * from "./icons/warning"
export * from "./icons/success"
export * from "./icons/reduce"
export * from "./icons/empty"
export * from "./icons/warning-circle"
export * from "./icons/info-cricle"
export * from "./icons/pre"
export * from "./icons/next"
export * from "./icons/up"
export * from "./icons/down"
export * from "./icons/more"
export * from "./icons/add"
export * from "./icons/expand"
export * from "./icons/upload"
export * from "./icons/minus"
export * from "./icons/plus"
export * from "./icons/delete"
export * from "./icons/file-word"
export * from "./icons/file-pdf"
export * from "./icons/file-default"
export * from "./icons/file-excel"
export * from "./icons/file-music"
export * from "./icons/file-ppt"
export * from "./icons/file-picture"
export * from "./icons/file-video"
export * from "./icons/file-wps"
export * from "./icons/like"
export * from "./icons/share"
export * from "./icons/star"
export * from "./icons/heart"
export * from "./icons/caret-down"
export * from "./icons/drag-point"
export * from "./icons/pen"
export * from "./icons/pre-double"
export * from "./icons/next-double"
export * from "./icons/leaf"
export * from "./icons/caret-left"
export * from "./icons/caret-right"
export * from "./icons/sorter-default"
export * from "./icons/sorter-up"
export * from "./icons/sorter-down"
export * from "./icons/filter"
export * from "./icons/time"
export * from "./icons/calendar"
export * from "./icons/caret-down"
export * from "./icons/caret-left"
export * from "./icons/caret-right"
export * from "./icons/info"
