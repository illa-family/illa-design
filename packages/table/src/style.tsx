import { css, SerializedStyles } from "@emotion/react"
import { globalColor, illaPrefix } from "@illa-design/theme"
import { TableAlign, TableLayout, TableSize } from "./interface"

export function applySizeStyle(size: TableSize): SerializedStyles {
  let paddingStyle
  switch (size) {
    case "small":
      paddingStyle = css`
        padding: 5px 16px;
      `
      break
    case "medium":
      paddingStyle = css`
        padding: 7px 16px;
      `
      break
    case "large":
      paddingStyle = css`
        padding: 9px 16px;
      `
      break
  }
  return paddingStyle
}

export function applyHeaderSizeStyle(size: TableSize): SerializedStyles {
  let paddingStyle
  switch (size) {
    case "small":
      paddingStyle = css`
        padding: 5px 0;
      `
      break
    case "medium":
      paddingStyle = css`
        padding: 7px 0;
      `
      break
    case "large":
      paddingStyle = css`
        padding: 9px 0;
      `
      break
  }
  return paddingStyle
}

export function applyContainerStyle(_css?: SerializedStyles): SerializedStyles {
  return css(
    css`
      overflow: auto;
      display: inline-flex;
      vertical-align: center;
    `,
    _css,
  )
}

export function applyResizing(canResize?: boolean): SerializedStyles {
  if (canResize) {
    return css`
      width: 10px;
      height: 100%;
      z-index: 10;
      position: absolute;
      right: 0;
      transform: translateX(50%);
    `
  }
  return css``
}

export function applyBorderStyle(
  borderCell?: boolean,
  striped?: boolean,
): SerializedStyles {
  let borderStyle: SerializedStyles = css()
  if (borderCell) {
    borderStyle = css`
      border-left: solid 1px ${globalColor(`--${illaPrefix}-gray-08`)};
      border-right: solid 1px ${globalColor(`--${illaPrefix}-gray-08`)};
    `
  }
  let stripedStyle: SerializedStyles = css()
  if (striped) {
    stripedStyle = css`
      border-top: solid 1px ${globalColor(`--${illaPrefix}-gray-08`)};
      border-bottom: solid 1px ${globalColor(`--${illaPrefix}-gray-08`)};
    `
  }
  return css(borderStyle, stripedStyle)
}

export function applyThStyle(): SerializedStyles {
  return css`
    font-size: 14px;
    font-weight: bold;
    background-color: ${globalColor(`--${illaPrefix}-gray-09`)};
    color: ${globalColor(`--${illaPrefix}-gray-02`)};
  `
}

export const applyNormalStyle: SerializedStyles = css`
  font-size: 14px;
  min-height: 22px;
  background-color: ${globalColor(`--${illaPrefix}-white-01`)};
  color: ${globalColor(`--${illaPrefix}-gray-02`)};
`

export const applyFilterContainer = css`
  flex-shrink: 1;
  display: flex;
  flex-direction: row;
  align-items: center;
`

export const applyHeaderIconLeft = css`
  margin-left: 4px;
  width: 16px;
  height: 16px;
  flex-wrap: wrap;
`

export function applyContentContainer(align: TableAlign): SerializedStyles {
  return css`
    justify-content: ${align};
    display: flex;
    overflow: auto;
    min-height: 22px;
    align-items: center;
    flex-direction: row;
  `
}

export function applyPreContainer(align: TableAlign): SerializedStyles {
  return css`
    justify-content: ${align};
    display: inline-flex;
    flex-shrink: 0;
    align-items: center;
    flex-direction: row;
    flex-grow: 1;
  `
}

export function applyTableStyle(
  tableLayout: TableLayout,
  bordered?: boolean,
): SerializedStyles {
  const border = css`
    border: solid 1px ${globalColor(`--${illaPrefix}-gray-08`)};
  `

  const commonStyle = css`
    width: 100%;
    box-sizing: border-box;
    display: table;
    border-collapse: collapse;
    table-layout: ${tableLayout};
  `

  if (bordered) {
    return css`
      ${border};
      ${commonStyle};
    `
  } else {
    return css`
      ${commonStyle};
    `
  }
}
