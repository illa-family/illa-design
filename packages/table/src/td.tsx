import { forwardRef, useContext } from "react"
import { TdProps } from "./interface"
import {
  applyBorderStyle,
  applyContentContainer,
  applyNormalStyle,
  applySizeStyle,
} from "./style"
import { css } from "@emotion/react"
import { TableContext } from "./table-context"

export const Td = forwardRef<HTMLTableDataCellElement, TdProps>(
  (props, ref) => {
    const {
      size,
      borderedCell,
      striped,
      align,
      children,
      showFooter,
      showHeader,
      _css,
      ...otherProps
    } = props

    const tableContext = useContext(TableContext)

    return (
      <td
        css={css(
          applyNormalStyle,
          applySizeStyle(size ?? tableContext?.size ?? "medium"),
          applyBorderStyle(
            borderedCell ?? tableContext?.borderedCell,
            striped ?? tableContext?.striped,
          ),
          _css,
        )}
        ref={ref}
        {...otherProps}
      >
        <div
          css={applyContentContainer(align ?? tableContext?.align ?? "left")}
        >
          {children}
        </div>
      </td>
    )
  },
)

Td.displayName = "Td"
