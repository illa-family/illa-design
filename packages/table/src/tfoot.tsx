import { forwardRef, useContext } from "react"
import { TFootProps } from "./interface"
import { css } from "@emotion/react"
import { TableContext } from "./table-context"

export const TFoot = forwardRef<HTMLTableSectionElement, TFootProps>(
  (props, ref) => {
    const { _css, ...otherProps } = props
    const tableContext = useContext(TableContext)
    return tableContext?.showFooter ? (
      <tfoot css={css(_css)} ref={ref} {...otherProps} />
    ) : null
  },
)

TFoot.displayName = "TFoot"
