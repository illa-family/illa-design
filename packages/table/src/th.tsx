import * as React from "react"
import { forwardRef, useContext, useEffect, useRef, useState } from "react"
import { ThProps } from "./interface"
import {
  applyBorderStyle,
  applyContentContainer,
  applySizeStyle,
  applyThStyle,
} from "./style"
import { css } from "@emotion/react"
import { TableContext } from "./table-context"
import { mergeRefs } from "@illa-design/system"

export const Th = forwardRef<HTMLTableHeaderCellElement, ThProps>(
  (props, ref) => {
    const {
      size,
      borderedCell,
      striped,
      align,
      children,
      showFooter,
      showHeader,
      _css,
      ...otherProps
    } = props
    const tableContext = useContext(TableContext)

    const thRef = useRef<HTMLTableHeaderCellElement | null>()
    const [thTop, setThTop] = useState(0)

    useEffect(() => {
      setThTop(thRef.current?.parentElement?.offsetTop ?? 0)
    })

    return (
      <th
        css={css(
          applyThStyle(),
          applySizeStyle(size ?? tableContext?.size ?? "medium"),
          applyBorderStyle(
            borderedCell ?? tableContext?.borderedCell,
            striped ?? tableContext?.striped,
          ),
          _css,
        )}
        ref={mergeRefs(ref, thRef)}
        {...otherProps}
      >
        <div
          css={applyContentContainer(align ?? tableContext?.align ?? "left")}
        >
          {children}
        </div>
      </th>
    )
  },
)

Th.displayName = "Th"
