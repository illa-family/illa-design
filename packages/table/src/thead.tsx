import { forwardRef, useContext } from "react"
import { THeadProps } from "./interface"
import { css } from "@emotion/react"
import { TableContext } from "./table-context"

export const Thead = forwardRef<HTMLTableSectionElement, THeadProps>(
  (props, ref) => {
    const { _css, ...otherProps } = props
    const tableContext = useContext(TableContext)
    return tableContext?.showHeader ? (
      <thead css={css(_css)} ref={ref} {...otherProps} />
    ) : null
  },
)

Thead.displayName = "Thead"
