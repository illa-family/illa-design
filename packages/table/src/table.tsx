import {
  ChangeEvent,
  ForwardedRef,
  forwardRef,
  ReactElement,
  SyntheticEvent,
  useEffect,
} from "react"
import { TableContextProps, TableProps } from "./interface"
import {
  applyContainerStyle,
  applyFilterContainer,
  applyHeaderIconLeft,
  applyPreContainer,
  applyResizing,
  applyTableStyle,
} from "./style"
import { TableContext } from "./table-context"
import {
  Row,
  useFilters,
  useResizeColumns,
  useRowSelect,
  useSortBy,
  useTable,
} from "react-table"
import { Thead } from "./thead"
import { Tr } from "./tr"
import { Th } from "./th"
import { TBody } from "./tbody"
import { Td } from "./td"
import { TFoot } from "./tfoot"
import {
  SorterDefaultIcon,
  SorterDownIcon,
  SorterUpIcon,
} from "@illa-design/icon"
import { Checkbox } from "@illa-design/checkbox"
import { TableData } from "./table-data"

export const Table = forwardRef<HTMLDivElement, TableProps<any>>(
  (props, ref) => {
    if (props.columns == undefined || props.data == undefined) {
      return renderDirectTable(props, ref)
    } else {
      return renderDataDrivenTable(props, ref)
    }
  },
)

function renderDirectTable<D extends TableData>(
  props: TableProps<D>,
  ref: ForwardedRef<HTMLDivElement>,
): ReactElement {
  const {
    size = "medium",
    tableLayout = "auto",
    columns,
    data,
    bordered,
    borderedCell,
    striped,
    children,
    disableSortBy,
    disableFilters,
    disableRowSelect,
    align = "left",
    showFooter = true,
    showHeader = true,
    onRowSelectChange,
    _css,
    ...otherProps
  } = props

  const contextProps = {
    align,
    borderedCell,
    striped,
    size,
    showHeader,
    showFooter,
  } as TableContextProps

  return (
    <div css={applyContainerStyle(_css)} ref={ref}>
      <TableContext.Provider value={contextProps}>
        <table css={applyTableStyle(tableLayout, bordered)} {...otherProps}>
          {children}
        </table>
      </TableContext.Provider>
    </div>
  )
}

function renderDataDrivenTable<D extends TableData>(
  props: TableProps<D>,
  ref: ForwardedRef<HTMLDivElement>,
): ReactElement {
  const {
    size = "medium",
    tableLayout = "auto",
    columns = [],
    data = [],
    bordered,
    borderedCell,
    striped,
    children,
    disableSortBy,
    disableFilters,
    disableRowSelect,
    align = "left",
    showFooter,
    showHeader = true,
    onRowSelectChange,
    _css,
    ...otherProps
  } = props

  const contextProps = {
    align,
    borderedCell,
    striped,
    size,
    showHeader,
    showFooter,
  } as TableContextProps

  const {
    headerGroups,
    rows,
    prepareRow,
    footerGroups,
    getTableProps,
    getTableBodyProps,
    selectedFlatRows,
    allColumns,
  } = useTable<D>(
    {
      columns,
      data,
      disableSortBy,
      disableFilters,
      // TODO resizing
      disableResizing: true,
    },
    useFilters,
    useSortBy,
    useResizeColumns,
    useRowSelect,
    (hooks) => {
      // add selection
      hooks.visibleColumns.push((columns) => [
        {
          id: "selection",
          Header: ({ getToggleAllRowsSelectedProps }) => (
            <Checkbox
              checked={getToggleAllRowsSelectedProps().checked}
              indeterminate={getToggleAllRowsSelectedProps().indeterminate}
              onChange={(_, event: SyntheticEvent) => {
                let changeEvent = getToggleAllRowsSelectedProps().onChange
                if (changeEvent != undefined) {
                  changeEvent({
                    target: event.target,
                  } as ChangeEvent)
                }
              }}
            />
          ),
          disableSortBy: true,
          disableResizing: true,
          width: "0px",
          Cell: ({ row }: { row: Row<D> }) => (
            <Checkbox
              disabled={row.original.disableRowSelect}
              checked={row.getToggleRowSelectedProps().checked}
              indeterminate={row.getToggleRowSelectedProps().indeterminate}
              onChange={(checked: boolean, event: SyntheticEvent) => {
                let changeEvent = row.getToggleRowSelectedProps().onChange
                if (changeEvent != undefined) {
                  changeEvent({
                    target: event.target,
                  } as ChangeEvent)
                }
              }}
            />
          ),
        },
        ...columns,
      ])
    },
  )

  // row select event
  if (onRowSelectChange != undefined) {
    onRowSelectChange(selectedFlatRows)
  }

  useEffect(() => {
    // hidden select
    allColumns.map((column) => {
      if (disableRowSelect && column.id == "selection") {
        column.toggleHidden(true)
      } else {
        column.toggleHidden(false)
      }
    })
  }, [disableRowSelect])

  return (
    <div css={applyContainerStyle(_css)} ref={ref}>
      <TableContext.Provider value={contextProps}>
        <table
          css={applyTableStyle(tableLayout, bordered)}
          {...getTableProps()}
          {...otherProps}
        >
          {showHeader && (
            <Thead>
              {headerGroups.map((group) => (
                <Tr {...group.getHeaderGroupProps()}>
                  {group.headers.map((column, index) => (
                    <Th {...column.getHeaderProps()}>
                      <div
                        css={applyPreContainer(align)}
                        {...column.getSortByToggleProps()}
                      >
                        {column.Header != undefined && column.render("Header")}
                        {column.canSort &&
                          (column.isSorted ? (
                            column.isSortedDesc ? (
                              <SorterDownIcon _css={applyHeaderIconLeft} />
                            ) : (
                              <SorterUpIcon _css={applyHeaderIconLeft} />
                            )
                          ) : (
                            <SorterDefaultIcon _css={applyHeaderIconLeft} />
                          ))}
                      </div>
                      <div css={applyFilterContainer}>
                        {column.canFilter &&
                          column.Filter != undefined &&
                          column.render("Filter")}
                      </div>
                      {column.canResize &&
                        group.headers.indexOf(column) !=
                          group.headers.length - 1 && (
                          <div
                            {...column.getResizerProps()}
                            css={applyResizing(column.canResize)}
                          />
                        )}
                    </Th>
                  ))}
                </Tr>
              ))}
            </Thead>
          )}
          <TBody {...getTableBodyProps()}>
            {rows.map((row) => {
              prepareRow(row)
              return (
                <Tr {...row.getRowProps()}>
                  {row.cells.map((cell, index) => {
                    return (
                      <Td {...cell.getCellProps()}>{cell.render("Cell")}</Td>
                    )
                  })}
                </Tr>
              )
            })}
          </TBody>
          {showFooter && (
            <TFoot>
              {footerGroups.map((group) => (
                <Tr {...group.getFooterGroupProps()}>
                  {group.headers.map((column, index) => (
                    <Td {...column.getFooterProps()}>
                      {column.Footer != undefined && column.render("Footer")}
                    </Td>
                  ))}
                </Tr>
              ))}
            </TFoot>
          )}
        </table>
      </TableContext.Provider>
    </div>
  )
}

Table.displayName = "Table"
