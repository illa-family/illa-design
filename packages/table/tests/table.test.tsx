import { Table, TBody, Td, TFoot, Th, Thead, Tr } from "../src"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import { globalColor, illaPrefix } from "@illa-design/theme"

test("Table renders correctly.", () => {
  render(
    <Table data-testid="test-table">
      <Thead>
        <Tr>
          <Th>First</Th>
          <Th>Second</Th>
          <Th>Third</Th>
        </Tr>
      </Thead>
      <TBody>
        <Tr>
          <Td>1</Td>
          <Td>2</Td>
          <Td>3</Td>
        </Tr>
        <Tr>
          <Td>4</Td>
          <Td>5</Td>
          <Td>6</Td>
        </Tr>
      </TBody>
      <TFoot>
        <Tr>
          <Td>7</Td>
          <Td>8</Td>
          <Td>9</Td>
        </Tr>
      </TFoot>
    </Table>,
  )
  expect(screen.getByTestId("test-table")).toBeInTheDocument()
})

test("Table renders with borders.", () => {
  render(
    <Table data-testid="test-table" bordered borderedCell striped>
      <Thead>
        <Tr>
          <Th data-testid="test-th">First</Th>
          <Th>Second</Th>
          <Th>Third</Th>
        </Tr>
      </Thead>
      <TBody>
        <Tr>
          <Td data-testid="test-td">1</Td>
          <Td>2</Td>
          <Td>3</Td>
        </Tr>
        <Tr>
          <Td>4</Td>
          <Td>5</Td>
          <Td>6</Td>
        </Tr>
      </TBody>
      <TFoot>
        <Tr>
          <Td>7</Td>
          <Td>8</Td>
          <Td>9</Td>
        </Tr>
      </TFoot>
    </Table>,
  )
  const itemCss = {
    borderLeft: `solid 1px ${globalColor(`--${illaPrefix}-gray-08`)}`,
    borderRight: `solid 1px ${globalColor(`--${illaPrefix}-gray-08`)}`,
    borderTop: `solid 1px ${globalColor(`--${illaPrefix}-gray-08`)}`,
    borderBottom: `solid 1px ${globalColor(`--${illaPrefix}-gray-08`)}`,
  }
  expect(screen.getByTestId("test-table")).toHaveStyle({
    border: `solid 1px ${globalColor(`--${illaPrefix}-gray-08`)}`,
  })
  expect(screen.getByTestId("test-th")).toHaveStyle(itemCss)
  expect(screen.getByTestId("test-td")).toHaveStyle(itemCss)
})

test("Table renders without header.", () => {
  render(
    <Table showHeader={false}>
      <Thead data-testid="test-header">
        <Tr>
          <Th>First</Th>
          <Th>Second</Th>
          <Th>Third</Th>
        </Tr>
      </Thead>
      <TBody>
        <Tr>
          <Td>1</Td>
          <Td>2</Td>
          <Td>3</Td>
        </Tr>
        <Tr>
          <Td>4</Td>
          <Td>5</Td>
          <Td>6</Td>
        </Tr>
      </TBody>
      <TFoot>
        <Tr>
          <Td>7</Td>
          <Td>8</Td>
          <Td>9</Td>
        </Tr>
      </TFoot>
    </Table>,
  )
  expect(screen.queryByTestId("test-header")).not.toBeInTheDocument()
})

test("Table renders without footer.", () => {
  render(
    <Table showFooter={false}>
      <Thead data-testid="test-header">
        <Tr>
          <Th>First</Th>
          <Th>Second</Th>
          <Th>Third</Th>
        </Tr>
      </Thead>
      <TBody>
        <Tr>
          <Td>1</Td>
          <Td>2</Td>
          <Td>3</Td>
        </Tr>
        <Tr>
          <Td>4</Td>
          <Td>5</Td>
          <Td>6</Td>
        </Tr>
      </TBody>
      <TFoot data-testid="test-footer">
        <Tr>
          <Td>7</Td>
          <Td>8</Td>
          <Td>9</Td>
        </Tr>
      </TFoot>
    </Table>,
  )
  expect(screen.queryByTestId("test-footer")).not.toBeInTheDocument()
})

test("Table renders with different table layout.", () => {
  render(
    <Table tableLayout="fixed" data-testid="test-table">
      <Thead>
        <Tr>
          <Th>First</Th>
          <Th>Second</Th>
          <Th>Third</Th>
        </Tr>
      </Thead>
      <TBody>
        <Tr>
          <Td>1</Td>
          <Td>2</Td>
          <Td>3</Td>
        </Tr>
      </TBody>
    </Table>,
  )
  expect(screen.queryByTestId("test-table")).toHaveStyle({
    tableLayout: "fixed",
  })
})

test("Table renders with different size.", () => {
  render(
    <Table size="large">
      <TBody>
        <Tr>
          <Td data-testid="test-td">1</Td>
          <Td>2</Td>
          <Td>3</Td>
        </Tr>
      </TBody>
    </Table>,
  )
  expect(screen.queryByTestId("test-td")).toHaveStyle({
    padding: "9px 16px",
  })
})

test("Table renders with different align.", () => {
  render(
    <Table align="center" data-testid="test-table">
      <TBody>
        <Tr>
          <Td>1</Td>
          <Td align="end">2</Td>
          <Td>3</Td>
        </Tr>
      </TBody>
    </Table>,
  )
  expect(screen.queryByTestId("test-table")).toMatchSnapshot()
})
