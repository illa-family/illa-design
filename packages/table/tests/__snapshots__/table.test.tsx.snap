// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders with different align. 1`] = `
.emotion-0 {
  width: 100%;
  box-sizing: border-box;
  display: table;
  border-collapse: collapse;
  table-layout: auto;
}

.emotion-3 {
  font-size: 14px;
  min-height: 22px;
  background-color: #fff;
  color: #1f1f1f;
  padding: 7px 16px;
}

.emotion-4 {
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: auto;
  min-height: 22px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-6 {
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: end;
  justify-content: end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: auto;
  min-height: 22px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<table
  class="emotion-0"
  data-testid="test-table"
>
  <tbody
    class="emotion-1"
  >
    <tr
      class="emotion-1"
    >
      <td
        class="emotion-3"
      >
        <div
          class="emotion-4"
        >
          1
        </div>
      </td>
      <td
        class="emotion-3"
      >
        <div
          class="emotion-6"
        >
          2
        </div>
      </td>
      <td
        class="emotion-3"
      >
        <div
          class="emotion-4"
        >
          3
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
